Move ubuntu-image.rst to Ubuntu Install Guide

rkratky opened the pull request:
This PR removes the `ubuntu-image.rst` file (the former man page) because it was moved to be published as a reference page in the [Ubuntu Installation Guide](https://canonical-subiquity.readthedocs-hosted.com/en/latest/index.html).

See https://github.com/canonical/subiquity/pull/1929 for details (do not merge this PR until the linked one is merged -- only then the new link will become available).

upils commented on README.md:
```diff
@@ -28,9 +28,8 @@ Alternatively, install the packages named in the `Build-Depends` field.
-* Manual page: [`man ubuntu-image`](https://github.com/Canonical/ubuntu-image/blob/main/ubuntu-image.rst)
-
-The `gadget.yaml` specification has moved to [the snapcraft forum](https://forum.snapcraft.io/t/gadget-snaps).
+* Reference page: [`ubuntu-image` syntax and options](https://canonical-subiquity.readthedocs-hosted.com/en/latest/reference/reference/ubuntu-image.html)
```
I cannot test because the doc is not yet merged but are you sure about having 2 "reference" in the link?

rkratky replied:
Good catch. Thanks.

upils reviewed (commented):
LGTM.

Do not worry about the dependencies vuln check failing, this is unrelated and will be fixed in another PR.

upils commented:
I think you should also remove this check https://github.com/canonical/ubuntu-image/blob/main/.github/workflows/documentation.yml#L51 and the next one, that will not make sense anymore.

rkratky commented:
> I think you should also remove this check https://github.com/canonical/ubuntu-image/blob/main/.github/workflows/documentation.yml#L51 and the next one, that will not make sense anymore.

Good point.

I have fixed the link and removed the two checks.
